File: application/src/main/kotlin/br/com/dillmann/nginxignition/application/common/pagination/PageResponse.kt
package br.com.dillmann.nginxignition.application.common.pagination

import kotlinx.serialization.Serializable

@Serializable
data class PageResponse<T>(
    val pageSize: Int,
    val pageNumber: Int,
    val totalItems: Long,
    val contents: List<T>,
)

File: frontend/src/core/pagination/PageResponse.ts
export function emptyPageResponse<T>(): PageResponse<T> {
    return {
        pageSize: 0,
        pageNumber: 0,
        totalItems: 0,
        contents: [],
    }
}

export default interface PageResponse<T> {
    pageSize: number
    pageNumber: number
    totalItems: number
    contents: T[]
}
